### ISSUE TITLE
workspace picture not update in report screen

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#19550]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. go to Settings > Workspaces
2. create new workspace
3. close the modal
4. click on workspace chat in LHN
5. go to Settings > Workspaces
6. change the workspace picture
7. check the workspace picture in report screen

## Expected Result: 
should show updated workspace picture in report screen
## Actual Result: 
does not show updated workspace picture in report screen

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.17-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/20d696a3-0503-4de9-be64-156c5b933941

https://github.com/Expensify/App/assets/43996225/2960b16a-ddbe-413a-81fa-f62f1722797d

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1684772113159839?thread_ts=1684126449.749049&cid=C049HHMV9SM

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by hungvu193** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace picture not update in report screen
### What is the root cause of that problem?
Problem came from here:
https://github.com/Expensify/App/blob/7a42755a79737314edf702ce17bf0647dd8c41fb/src/pages/home/report/ReportActionsView.js#L172
Since the `icons` doesn't exist in `this.props.report` this will make this condition always return false. That's why when we updated the avatar, the avatar of `ReportActionItemCreated` isn't updated. 
### What changes do you think we should make in order to solve the problem?
Add the icons props to `ReportActionsView` and change the condition of `shouldComponentUpdate`:
```javascript
const currentReportData = (isTaskReport || !isThread) && props.report.parentReportID ? props.parentReport : props.report;
const icons = ReportUtils.getIcons(this.props.report, this.props.personalDetails);

                                <ReportActionsView
                                    reportActions={this.props.reportActions}
                                    report={this.props.report}
                                    isComposerFullSize={this.props.isComposerFullSize}
                                    isDrawerOpen={this.props.isDrawerOpen}
                                    parentViewHeight={this.state.skeletonViewContainerHeight}
                                    personalDetails={this.props.personalDetails}
                                    icons={icons}
                                />
```

and the condition of shouldComponentUpdate inside `ReportActionsView`:

```javascript
return !_.isEqual(lodashGet(this.props, 'icons', []), lodashGet(nextProps, 'icons', []));
```

### What alternative solutions did you explore? (Optional)
N/A

###Result

https://github.com/Expensify/App/assets/16502320/23dae7e9-3879-492f-9b8e-a47a2f7b5fcf

**Comment by hungvu193** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace picture not update in report screen
### What is the root cause of that problem?
Problem came from here:
https://github.com/Expensify/App/blob/7a42755a79737314edf702ce17bf0647dd8c41fb/src/pages/home/report/ReportActionsView.js#L172
Since the `icons` doesn't exist in `this.props.report` this will make this condition always return false. That's why when we updated the avatar, the avatar of `ReportActionItemCreated` isn't updated. 
### What changes do you think we should make in order to solve the problem?
Add the icons props to `ReportActionsView` and change the condition of `shouldComponentUpdate`:
```javascript
const currentReportData = (isTaskReport || !isThread) && props.report.parentReportID ? props.parentReport : props.report;
const icons = ReportUtils.getIcons(this.props.report, this.props.personalDetails);

                                <ReportActionsView
                                    reportActions={this.props.reportActions}
                                    report={this.props.report}
                                    isComposerFullSize={this.props.isComposerFullSize}
                                    isDrawerOpen={this.props.isDrawerOpen}
                                    parentViewHeight={this.state.skeletonViewContainerHeight}
                                    personalDetails={this.props.personalDetails}
                                    icons={icons}
                                />
```

and the condition of shouldComponentUpdate inside `ReportActionsView`:

```javascript
return !_.isEqual(lodashGet(this.props, 'icons', []), lodashGet(nextProps, 'icons', []));
```
We can keep the old condition above the new condition if we want but we need to change it to` if ` condition:

```javascript
if (!isEqual(lodashGet(this.props.report, 'icons', []), lodashGet(nextProps.report, 'icons', [])) {
return true;
}
```

### What alternative solutions did you explore? (Optional)
In case we don't wanna pass a new props, we can also calculate the icons inside shouldComponentUpdate of `ReportActionsView`:
```javascript
  const icons = ReportUtils.getIcons(this.props.report, this.props.personalDetails);
  const nextIcons = ReportUtils.getIcons(nextProps, this.props.personalDetails);



return !_.isEqual(icons, nextIcons);
```
### Result

https://github.com/Expensify/App/assets/16502320/23dae7e9-3879-492f-9b8e-a47a2f7b5fcf

**Comment by abdulrahuman5196** [comment]
Reviewing this now

**Comment by abdulrahuman5196** [comment]
@hungvu193's Root cause is correct and the proposal is also working fine while testing.
I like the alternate solution since I think its better to not create a new param just to solve this issue.

🎀👀🎀
C+ Reviewed

cc: @iwiznia

**Comment by abdulrahuman5196** [comment]
@hungvu193  Just a minor note, in the alternate solution `const nextIcons = ReportUtils.getIcons(nextProps, this.props.personalDetails);` should have nextProps.report

**Comment by iwiznia** [comment]
I also like the alternative.

**Comment by hungvu193** [comment]
https://github.com/Expensify/App/pull/19750 is ready for review!

**Comment by twisterdotcom** [comment]
Nice, just waiting on responses to @abdulrahuman5196 in the PR @hungvu193

**Comment by abdulrahuman5196** [comment]
Bugzero checklist:

>The PR that introduced the bug has been identified. Link to the PR:
The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

This isn't a regression. But a miss from the original implementation and we didn't get to know of this issue until now.

>A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
[@abdulrahuman5196] Determine if we should create a regression test for this bug.
[@abdulrahuman5196] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

Very minor case only present in large screen devices. So I don't think regression test is beneficial in this case.

**Comment by twisterdotcom** [comment]
Okay, all offers sent. Sorry for the delay.

**Comment by gadhiyamanan** [comment]
@twisterdotcom  accepted, Thanks!

**Comment by abdulrahuman5196** [comment]
@twisterdotcom  accepted

**Comment by twisterdotcom** [comment]
Wow, you're all online! Great. Okay, we didn't get the "bonus" issue comment, but just double checking, there isn't one due.

Assigned: https://github.com/Expensify/App/issues/19550#event-9370255567 - May 29th, 2023 @ 4:38 PM GMT +1
Merged: https://github.com/Expensify/App/pull/19750#event-9436231511 - June 5, 2023 @ 6.12 PM GMT +1

**Comment by twisterdotcom** [comment]
I think if this solved another issue too, your "bonus" is the payment for that issue as well.
